lib/platform.cpp

```diff
@@ -200,20 +192,33 @@ bool Platform::loadFromFile(const char exename[], const std::string &filename, b
         fullfilename += ".xml";
 
     // TODO: use native separators
-    std::vector<std::string> filenames{
-        fullfilename,
-    };
-    if (!is_abs_path) {
-        filenames.push_back("platforms/" + fullfilename);
-        if (exename && (std::string::npos != Path::fromNativeSeparators(exename).find('/'))) {
-            filenames.push_back(Path::getPathFromFilename(Path::fromNativeSeparators(exename)) + fullfilename);
-            filenames.push_back(Path::getPathFromFilename(Path::fromNativeSeparators(exename)) + "platforms/" + fullfilename);
+    std::vector<std::string> filenames;
+    if (is_abs_path)
+    {
+        filenames.push_back(fullfilename);
+    }
+    else {
+        // TODO: drop duplicated paths
+        for (const std::string& path : paths)
+        {
+            if (path.empty())
+                continue; // TODO: error out instead?
+
+            std::string ppath = Path::fromNativeSeparators(path);
+            if (ppath.back() != '/')
+                ppath += '/';
+            // TODO: look in platforms first?
+            filenames.push_back(ppath + fullfilename);
+            filenames.push_back(ppath + "platforms/" + fullfilename);
         }
 #ifdef FILESDIR
         std::string filesdir = FILESDIR;
-        if (!filesdir.empty() && filesdir[filesdir.size()-1] != '/')
-            filesdir += '/';
-        filenames.push_back(filesdir + ("platforms/" + fullfilename));
+        if (!filesdir.empty()) {
+            if (filesdir.back() != '/')
+                filesdir += '/';
+            // TODO: look in filesdir?
```

Comment:
If we would also look into that folder it could just be one of the lookup paths above.